lib/sqlalchemy/dialects/sqlite/base.py

```diff
@@ -2651,7 +2667,38 @@ def get_indexes(self, connection, table_name, schema=None, **kw):
                 "sqlite_autoindex"
             ):
                 continue
-            indexes.append(dict(name=row[1], column_names=[], unique=row[2]))
+            indexes.append(
+                dict(
+                    name=row[1],
+                    column_names=[],
+                    unique=row[2],
+                    dialect_options={},
+                )
+            )
+
+            # check partial indexes
+            if row[4]:
+                s = (
+                    "SELECT sql FROM %(schema)ssqlite_master "
+                    "WHERE name = ? "
+                    "AND type = 'index'" % {"schema": schema_expr}
+                )
+                rs = connection.exec_driver_sql(s, (row[1],))
+                try:
+                    # in theory the code below shouldn't happen because
+                    # we know this is a partial index, so the definition
+                    # sql should be there and match the regex
+                    index_sql = rs.scalar()
+                    predicate = partial_pred_re.search(index_sql).group(1)
+                    indexes[-1]["dialect_options"]["sqlite_where"] = text(
+                        predicate
+                    )
+                except Exception:
+                    util.warn(
+                        "Failed to look up filter predicate of "
+                        "partial index %s" % row[1]
+                    )
+                    indexes.pop()
```

Comment:
**Federico Caselli** (CaselIT) wrote:

this is a change compared to how the dialect works now (since it currently returns partial indexes without informations instead of ignoring them if it cannot parse them)

but since it's in an edge case that should never happen, at least when used with sqlite, I think it's fine

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4200